community/cypher/cypher/src/main/scala/org/neo4j/cypher/CypherException.scala

```diff
@@ -152,3 +152,5 @@ class LoadExternalResourceException(message: String, cause: Throwable) extends C
 class LoadCsvStatusWrapCypherException(extraInfo: String, cause: CypherException) extends CypherException(s"${cause.getMessage} (${extraInfo})", cause) {
   val status = cause.status
 }
+
+class ShortestPathFallbackDisabledException(message: String, cause: Throwable) extends CypherExecutionException(message, cause)
```

Comment:
I'm not completely comfortable with the name of this class. It seems to focus on 'disabling the fallback', while I think the current focus is more on 'disallowing a potentially expensive exhaustive search'. The conditions under which this 'exception' could arise are:

* There is an existential or full-path predicate
* ShortestPath() failed to find anything

This combination leads to the suspicion that we have missed possible shortestPaths. Perhaps we want an error describing that? Hmmmm...... everything I'm coming up with is even worse.....